From 0f2f4616f2810ac6821ac8f85952874f422007c8 Mon Sep 17 00:00:00 2001
From: Emilio Pozuelo Monfort <pochu27@gmail.com>
Date: Sat, 12 Mar 2011 22:09:11 +0000
Subject: [PATCH] Run gio tests with a dbus session

Some gio tests (e.g. gapplication) need a running dbus session
to work, so we run them with dbus-launch, which prepares the
dbus session.
---
 Makefile.decl         |    2 +-
 gio/tests/Makefile.am |    3 +++
 2 files changed, 4 insertions(+), 1 deletions(-)

Index: glib-2.33.12/Makefile.decl
===================================================================
--- glib-2.33.12.orig/Makefile.decl	2012-08-25 13:14:10.000000000 +1200
+++ glib-2.33.12/Makefile.decl	2012-09-04 14:42:52.872116736 +1200
@@ -1,7 +1,7 @@
 # GLIB - Library of useful C routines
 
 #GTESTER = gtester 			# for non-GLIB packages
-GTESTER = $(top_builddir)/glib/gtester			# for the GLIB package
+GTESTER = $(GTESTER_ENV) $(top_builddir)/glib/gtester	# for the GLIB package
 GTESTER_REPORT = $(top_builddir)/glib/gtester-report	# for the GLIB package
 
 # initialize variables for unconditional += appending
Index: glib-2.33.12/gio/tests/Makefile.am
===================================================================
--- glib-2.33.12.orig/gio/tests/Makefile.am	2012-09-04 02:24:05.000000000 +1200
+++ glib-2.33.12/gio/tests/Makefile.am	2012-09-04 14:42:52.872116736 +1200
@@ -6,6 +6,9 @@
 
 SUBDIRS = gdbus-object-manager-example
 
+# Some tests need a dbus session
+GTESTER_ENV = dbus-launch
+
 INCLUDES = 			\
 	-DG_LOG_DOMAIN=\"GLib-GIO\" \
 	$(gio_INCLUDES)		\
